core/array.cpp

```diff
@@ -118,16 +118,25 @@ bool Array::operator==(const Array &p_array) const {
 }
 
 uint32_t Array::hash() const {
+	return recursive_hash(0);
+}
+
+uint32_t Array::recursive_hash(int p_recursion_count) const {
```

Comment:
Let me suggest declaring these as (private) `uint32_t T::_hash(int p_recursion_count = 0) const`.

Reply:
Makes sense. I just copied it from 4.x though.